src/DotLiquid/WhereFilter.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DotLiquid
+{
+    internal static class WhereFilter
+    {
+        private static readonly IEqualityComparer<object> Comparer = new ValueEqualityComparer();
+
+        public static IEnumerable<object> Where(object any, string key, object value)
+        {
+            return ToEnum(any)
+                .Where(el => Comparer.Equals(GetFieldVal(el, key), value));
+        }
+
+        private static IEnumerable<object> ToEnum(object any)
+        {
+            return any switch {
```

Comment:
There is some non obvious logic as to why these conversions need to be. Please add code comments and/or specification reference to explain.